goal_src/engine/gfx/lights.gc

```diff
@@ -5,3 +5,97 @@
 ;; name in dgo: lights
 ;; dgos: GAME, ENGINE
 
+(defun light-slerp ((out light) (a light) (b light) (alpha float))
+  "Linearly interpolate between two light's vectors. Alpha is clamped between 0 - 1."
+  (let ((clamped-alpha (fmax 0.0 (fmin 1.0 alpha))))
+   (vector-lerp! (-> out color) (-> a color) (-> b color) clamped-alpha)
+   (vector-deg-slerp
+    (-> out direction)
+    (-> a direction)
+    (-> b direction)
+    clamped-alpha
+    )
+   (let ((f0-2 (-> a levels x))
+         (f1-2 (-> b levels x))
+         )
+    (set! (-> out levels x) (+ f0-2 (* clamped-alpha (- f1-2 f0-2))))
+    )
+   )
+  out
+  )
+
+(defun
+  light-group-slerp
+  "Linearlyl interpolate between each of the 4 vector components of two light-groups.
```

Comment:
The docstring has to come after the arguments in a `defun`.

Also, I usually like to put all the arguments on the same line. Something like
```lisp
(defun light-group-slerp ((out light-group) (a light-group) (b light-group) (alpha float))
  "docstring"
   body
  )
```

If you use emacs, you can use alt + shift + 6 to quickly combine two lines.